src/components/biggive-tipping-slider/biggive-tipping-slider.tsx

```diff
@@ -32,7 +35,7 @@ export class BiggiveTippingSlider {
     var move = (e: MouseEvent | TouchEvent) => {
       if (isMoving) {
         const max = bar.offsetWidth - handle.offsetWidth;
-        const pageX = e instanceof TouchEvent ? e.touches[0]?.pageX : e.pageX;
+        const pageX = window.TouchEvent && e instanceof TouchEvent ? e.touches[0]?.pageX : (e as MouseEvent).pageX;
```

Comment:
@bdsl The workaround needed to stop e.g. desktop Firefox crashing is as per https://stackoverflow.com/a/32882849/2803757

If we apply _only_ that, then the last operand of the ternary gives a type error because touch events don't have `pageX`. `as MouseEvent` tells the type system what we already know from the workaround, and removes the error.

Reply:
OK - maybe this is a bit verbose, maybe it makes it clearer:


```suggestion
        let pageX: number | undefined;

        if (window.TouchEvent && e instanceof TouchEvent) {
          pageX = e.touches[0]?.pageX;
        } else {
          // we Know e is a MouseEvent because all platforms that supports TouchEvent would also have
          // a truthy window.TouchEvent - see https://stackoverflow.com/a/32882849/2803757
          pageX = (e as MouseEvent).pageX;
        }
```

Reply:
Yep, like it. Thanks!

Reply:
Thanks! The `as` always makes me nervous when I see it, so I wanted to have some comment around it at least.